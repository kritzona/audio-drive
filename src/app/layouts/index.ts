export { default as DashboardLayout } from './dashboard/DashboardLayout.vue';
