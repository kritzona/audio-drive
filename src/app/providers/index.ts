export { pinia } from './pinia';

export { vuetify } from './vuetify';
export { shallowVuetify } from './shallow-vuetify';

export { router } from './router';
