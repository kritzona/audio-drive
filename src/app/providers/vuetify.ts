import { createVuetify, ThemeDefinition } from 'vuetify';
import 'vuetify/styles';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';
import { aliases, mdi } from 'vuetify/iconsets/mdi';
import { md3 } from 'vuetify/blueprints';

const appTheme: ThemeDefinition = {
  dark: false,
  colors: {
    background: '#FFFFFF',
    surface: '#FFFFFF',
    primary: '#80489C',
    'primary-darken-1': '#432C7A',
    secondary: '#FF8FB1',
    'secondary-darken-1': '#FCE2DB',
    error: '#B00020',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00',
  },
};

export const vuetify = createVuetify({
  components,
  directives,
  icons: {
    defaultSet: 'mdi',
    aliases,
    sets: {
      mdi,
    },
  },
  blueprint: md3,
  theme: {
    defaultTheme: 'appTheme',
    themes: {
      appTheme,
    },
  },
});
