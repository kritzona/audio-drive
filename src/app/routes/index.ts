import { RouteRecordRaw } from 'vue-router';
import { Routes } from '@/shared/constants';

const routes: RouteRecordRaw[] = [
  {
    path: '/',
    name: Routes.MAIN_PAGE,
    component: () => import('@/pages/main/MainPage.vue'),
    beforeEnter: (_to, _from, next) => {
      next({ name: Routes.ALL_PLAYLIST_PAGE });
    },
  },
  {
    path: `/${Routes.UPLOAD_AUDIO_TRACK_PAGE}`,
    name: Routes.UPLOAD_AUDIO_TRACK_PAGE,
    component: () =>
      import('@/pages/upload-audio-track/UploadAudioTrackPage.vue'),
    meta: {
      title: 'Загрузка аудио-трека',
      icon: 'mdi-upload',
    },
  },
  {
    path: `/${Routes.ALL_PLAYLIST_PAGE}`,
    name: Routes.ALL_PLAYLIST_PAGE,
    component: () => import('@/pages/all-playlist/AllPlaylistPage.vue'),
    meta: {
      title: 'Все плейлисты',
      icon: 'mdi-playlist-music',
    },
  },
  {
    path: `/${Routes.ADD_PLAYLIST_PAGE}`,
    name: Routes.ADD_PLAYLIST_PAGE,
    component: () => import('@/pages/add-playlist/AddPlaylistPage.vue'),
    meta: {
      title: 'Добавление нового плейлиста',
      icon: 'mdi-playlist-plus',
    },
  },
];

export default routes;
