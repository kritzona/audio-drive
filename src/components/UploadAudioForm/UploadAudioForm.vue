<template>
  <v-card class="mx-auto px-6 py-8" max-width="344">
    <v-form class="upload-audio-form" @submit.prevent="handleSubmit">
      <v-text-field
        v-model="name"
        :readonly="loading"
        clearable
        hide-details="auto"
        label="Название"
        color="primary"
      />

      <v-text-field
        v-model="author"
        :readonly="loading"
        clearable
        hide-details="auto"
        label="Исполнитель"
        class="mt-5"
        color="primary"
      />

      <v-file-input
        v-model="covers"
        :readonly="loading"
        accept="image/png, image/jpeg, image/bmp"
        label="Обложка"
        color="primary"
        prepend-icon="mdi-camera"
        show-size
        class="mt-5"
      />

      <v-file-input
        v-model="audios"
        :readonly="loading"
        accept="audio/mpeg"
        label="Аудио-трек"
        color="primary"
        prepend-icon="mdi-music"
        show-size
      />

      <v-btn
        :loading="loading"
        block
        color="primary"
        size="large"
        type="submit"
        class="mt-5"
      >
        Загрузить
      </v-btn>
    </v-form>
  </v-card>
</template>

<script lang="ts" setup>
import { usePlayerStore } from '@/stores/player/player.store';
import { ref } from 'vue';
import { fileToBase64Url } from '@/utils/file.utils';

const playerStore = usePlayerStore();

const loading = ref<boolean>(false);

const name = ref<string>('');
const author = ref<string>('');

const covers = ref<File[]>([]);
const audios = ref<File[]>([]);

const handleSubmit = async () => {
  const [coverFile] = covers.value;
  const [audioFile] = audios.value;

  playerStore.setup({
    id: '1',
    name: name.value,
    author: author.value,
    cover: await fileToBase64Url(coverFile),
    fileName: audioFile.name,
    format: audioFile.type,
    url: await fileToBase64Url(audioFile),
  });
};
</script>
