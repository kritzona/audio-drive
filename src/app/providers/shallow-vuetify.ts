import { createVuetify } from 'vuetify';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';
import { md3 } from 'vuetify/blueprints';

export const shallowVuetify = createVuetify({
  components,
  directives,
  blueprint: md3,
});
