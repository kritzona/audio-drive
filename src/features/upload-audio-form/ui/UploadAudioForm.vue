<template>
  <v-form ref="form" class="upload-audio-form" @submit.prevent="handleSubmit">
    <v-text-field
      v-model="name"
      name="name"
      :readonly="loading"
      :rules="requiredTextRules"
      clearable
      hide-details="auto"
      label="Название"
      color="primary"
      required
    />

    <v-text-field
      v-model="author"
      name="author"
      :readonly="loading"
      :rules="requiredTextRules"
      clearable
      hide-details="auto"
      label="Исполнитель"
      class="mt-5"
      color="primary"
      required
    />

    <v-file-input
      :model-value="covers"
      name="covers"
      :readonly="loading"
      :rules="requiredFileRules"
      accept="image/png, image/jpeg, image/bmp"
      label="Обложка"
      color="primary"
      prepend-icon="mdi-camera"
      show-size
      class="mt-5"
      @update:model-value="handleCoversUpload"
    />

    <v-file-input
      :model-value="audios"
      name="audios"
      :readonly="loading"
      :rules="requiredFileRules"
      accept="audio/mpeg"
      label="Аудио-трек"
      color="primary"
      prepend-icon="mdi-music"
      show-size
      @update:model-value="handleAudiosUpload"
    />

    <v-btn
      :loading="loading"
      block
      color="primary"
      size="large"
      type="submit"
      class="mt-5"
    >
      Загрузить
    </v-btn>
  </v-form>
</template>

<script lang="ts" setup>
import { usePlayer } from '@/widgets/player';
import { ref } from 'vue';
import { fileToBase64Url } from '@/shared/lib';
import { VForm } from 'vuetify/components';
import { requiredFileRules, requiredTextRules } from '@/shared/constants';
import { useForm } from '@/shared/model';

/**
 * Бизнес-логика для работы с плеером
 */
const player = usePlayer();

/**
 * Бизнес-логика для работы с формой
 */
const { form, loading, validate } = useForm();

/**
 * Название трека
 */
const name = ref<string>('');
/**
 * Автор трека
 */
const author = ref<string>('');
/**
 * Файл обложки
 */
const covers = ref<File[]>([]);
/**
 * Аудио-файл
 */
const audios = ref<File[]>([]);

/**
 * Обработчик загрузки обложек
 *
 * @param files Файлы обложек
 */
const handleCoversUpload = (files: File[]) => (covers.value = files);

/**
 * Обработчик загрузки аудио-треков
 *
 * @param files Файлы аудио-треков
 */
const handleAudiosUpload = (files: File[]) => (audios.value = files);

/**
 * Обработчик отправки формы
 *
 * @async
 */
const handleSubmit = async () => {
  await validate();

  const [coverFile] = covers.value;
  const [audioFile] = audios.value;

  player.initTrack({
    id: '1',
    name: name.value,
    author: author.value,
    cover: await fileToBase64Url(coverFile),
    fileName: audioFile.name,
    format: audioFile.type,
    url: await fileToBase64Url(audioFile),
  });
};

defineExpose({
  form,
  loading,
  name,
  author,
  covers,
  audios,

  handleCoversUpload,
  handleAudiosUpload,
  handleSubmit,
});
</script>
