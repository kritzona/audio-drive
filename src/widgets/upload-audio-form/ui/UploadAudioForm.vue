<template>
  <v-form ref="form" class="upload-audio-form" @submit.prevent="handleSubmit">
    <v-text-field
      v-model="name"
      :readonly="loading"
      :rules="requiredTextRules"
      clearable
      hide-details="auto"
      label="Название"
      color="primary"
      required
    />

    <v-text-field
      v-model="author"
      :readonly="loading"
      :rules="requiredTextRules"
      clearable
      hide-details="auto"
      label="Исполнитель"
      class="mt-5"
      color="primary"
      required
    />

    <v-file-input
      v-model="covers"
      :readonly="loading"
      :rules="requiredFileRules"
      accept="image/png, image/jpeg, image/bmp"
      label="Обложка"
      color="primary"
      prepend-icon="mdi-camera"
      show-size
      class="mt-5"
    />

    <v-file-input
      v-model="audios"
      :readonly="loading"
      :rules="requiredFileRules"
      accept="audio/mpeg"
      label="Аудио-трек"
      color="primary"
      prepend-icon="mdi-music"
      show-size
    />

    <v-btn
      :loading="loading"
      block
      color="primary"
      size="large"
      type="submit"
      class="mt-5"
    >
      Загрузить
    </v-btn>
  </v-form>
</template>

<script lang="ts" setup>
import { usePlayerStore } from '@/app/stores';
import { ref } from 'vue';
import { fileToBase64Url } from '@/shared/lib';
import { VForm } from 'vuetify/components';
import {
  requiredFileRules,
  requiredTextRules,
} from '@/constants/validation-rules.constants';
import { useForm } from '@/composables/form.composable';

/**
 * Хранилище данных плеера
 */
const playerStore = usePlayerStore();

/**
 * Бизнес-логика для работы с формой
 */
const { form, loading, validate } = useForm();

/**
 * Название трека
 */
const name = ref<string>('');
/**
 * Автор трека
 */
const author = ref<string>('');
/**
 * Файл обложки
 */
const covers = ref<File[]>([]);
/**
 * Аудио-файл
 */
const audios = ref<File[]>([]);

/**
 * Обработчик отправки формы
 *
 * @async
 */
const handleSubmit = async () => {
  await validate();

  const [coverFile] = covers.value;
  const [audioFile] = audios.value;

  playerStore.setup({
    id: '1',
    name: name.value,
    author: author.value,
    cover: await fileToBase64Url(coverFile),
    fileName: audioFile.name,
    format: audioFile.type,
    url: await fileToBase64Url(audioFile),
  });
};
</script>
