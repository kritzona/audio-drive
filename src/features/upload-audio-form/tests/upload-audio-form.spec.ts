import { describe, beforeEach, it, expect, vi } from 'vitest';
import * as providers from '@/app/providers';
import { mount } from '@vue/test-utils';
import UploadAudioForm from '../ui/UploadAudioForm.vue';
import { createTestingPinia } from '@pinia/testing';

describe('UploadAudioForm.vue', () => {
  let vuetify: any;

  beforeEach(() => {
    vuetify = providers.shallowVuetify;
  });

  it('render', () => {
    const wrapper = mount(UploadAudioForm, {
      global: {
        plugins: [vuetify, createTestingPinia()],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('validation failure', async () => {
    const wrapper = mount(UploadAudioForm, {
      global: {
        plugins: [vuetify, createTestingPinia()],
      },
    });

    await expect(wrapper.vm.handleSubmit()).rejects.toEqual(undefined);
  });

  it('successful validation', async () => {
    const wrapper = mount(UploadAudioForm, {
      global: {
        plugins: [vuetify, createTestingPinia()],
      },
    });

    vi.mock('@/shared/lib', () => ({
      fileToBase64Url: vi.fn(),
    }));

    vi.mock('@/widgets/player', () => {
      const usePlayer = () => {
        return {
          initTrack: vi.fn(),
        };
      };

      return { usePlayer };
    });

    await wrapper.find('input[name="name"]').setValue('Тестовая песня');
    await wrapper.find('input[name="author"]').setValue('Исполнитель');
    wrapper.vm.handleCoversUpload([new File([''], 'cover.png')]);
    wrapper.vm.handleAudiosUpload([new File([''], 'audio.mp3')]);

    expect(wrapper.vm.handleSubmit()).resolves.toEqual(undefined);
  });
});
