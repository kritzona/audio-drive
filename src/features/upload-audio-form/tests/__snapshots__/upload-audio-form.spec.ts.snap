// Vitest Snapshot v1

exports[`UploadAudioForm.vue > render 1`] = `
"<form class=\\"v-form upload-audio-form\\" novalidate=\\"\\">
  <div class=\\"v-input v-input--horizontal v-input--density-default v-text-field\\">
    <!---->
    <div class=\\"v-input__control\\">
      <div class=\\"v-field v-field--appended v-field--variant-filled v-theme--light\\" role=\\"textbox\\">
        <div class=\\"v-field__overlay\\"></div>
        <div class=\\"v-field__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-primary\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long bg-primary\\"></div>
                <div class=\\"v-progress-linear__indeterminate short bg-primary\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class=\\"v-field__field\\" data-no-activator=\\"\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-0\\">
            <!---->Название
          </label><label class=\\"v-label v-field-label\\" for=\\"input-0\\">
            <!---->Название
          </label>
          <!----><input name=\\"name\\" size=\\"1\\" type=\\"text\\" id=\\"input-0\\" required=\\"\\" class=\\"v-field__input\\">
          <!---->
        </div>
        <transition-stub name=\\"expand-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-field__clearable\\" style=\\"display: none;\\"><i class=\\"mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Clear Название\\"></i></div>
        </transition-stub>
        <div class=\\"v-field__append-inner\\">
          <!---->
          <!---->
        </div>
        <div class=\\"v-field__outline\\">
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"v-input v-input--horizontal v-input--density-default v-text-field mt-5\\">
    <!---->
    <div class=\\"v-input__control\\">
      <div class=\\"v-field v-field--appended v-field--variant-filled v-theme--light\\" role=\\"textbox\\">
        <div class=\\"v-field__overlay\\"></div>
        <div class=\\"v-field__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-primary\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long bg-primary\\"></div>
                <div class=\\"v-progress-linear__indeterminate short bg-primary\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class=\\"v-field__field\\" data-no-activator=\\"\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-2\\">
            <!---->Исполнитель
          </label><label class=\\"v-label v-field-label\\" for=\\"input-2\\">
            <!---->Исполнитель
          </label>
          <!----><input name=\\"author\\" size=\\"1\\" type=\\"text\\" id=\\"input-2\\" required=\\"\\" class=\\"v-field__input\\">
          <!---->
        </div>
        <transition-stub name=\\"expand-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-field__clearable\\" style=\\"display: none;\\"><i class=\\"mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Clear Исполнитель\\"></i></div>
        </transition-stub>
        <div class=\\"v-field__append-inner\\">
          <!---->
          <!---->
        </div>
        <div class=\\"v-field__outline\\">
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"v-input v-input--horizontal v-input--density-default v-file-input mt-5\\">
    <div class=\\"v-input__prepend\\">
      <!----><i class=\\"mdi-camera mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Обложка prepended action\\"></i>
    </div>
    <div class=\\"v-input__control\\">
      <div class=\\"v-field v-field--appended v-field--variant-filled v-theme--light\\" prepend-icon=\\"mdi-camera\\">
        <div class=\\"v-field__overlay\\"></div>
        <div class=\\"v-field__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-primary\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long bg-primary\\"></div>
                <div class=\\"v-progress-linear__indeterminate short bg-primary\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class=\\"v-field__field\\" data-no-activator=\\"\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-5\\">
            <!---->Обложка
          </label><label class=\\"v-label v-field-label\\" for=\\"input-5\\">
            <!---->Обложка
          </label><input type=\\"file\\" name=\\"covers\\" id=\\"input-5\\" accept=\\"image/png, image/jpeg, image/bmp\\">
          <div class=\\"v-field__input\\">
            <!---->
          </div>
        </div>
        <transition-stub name=\\"expand-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-field__clearable\\" style=\\"display: none;\\"><i class=\\"mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Clear Обложка\\"></i></div>
        </transition-stub>
        <div class=\\"v-field__append-inner\\">
          <!---->
          <!---->
        </div>
        <div class=\\"v-field__outline\\">
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <div class=\\"v-input__details\\">
      <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\">
        <!---->
      </transition-group-stub>
      <!---->
    </div>
  </div>
  <div class=\\"v-input v-input--horizontal v-input--density-default v-file-input\\">
    <div class=\\"v-input__prepend\\">
      <!----><i class=\\"mdi-music mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Аудио-трек prepended action\\"></i>
    </div>
    <div class=\\"v-input__control\\">
      <div class=\\"v-field v-field--appended v-field--variant-filled v-theme--light\\" prepend-icon=\\"mdi-music\\">
        <div class=\\"v-field__overlay\\"></div>
        <div class=\\"v-field__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-primary\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long bg-primary\\"></div>
                <div class=\\"v-progress-linear__indeterminate short bg-primary\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class=\\"v-field__field\\" data-no-activator=\\"\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-7\\">
            <!---->Аудио-трек
          </label><label class=\\"v-label v-field-label\\" for=\\"input-7\\">
            <!---->Аудио-трек
          </label><input type=\\"file\\" name=\\"audios\\" id=\\"input-7\\" accept=\\"audio/mpeg\\">
          <div class=\\"v-field__input\\">
            <!---->
          </div>
        </div>
        <transition-stub name=\\"expand-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-field__clearable\\" style=\\"display: none;\\"><i class=\\"mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Clear Аудио-трек\\"></i></div>
        </transition-stub>
        <div class=\\"v-field__append-inner\\">
          <!---->
          <!---->
        </div>
        <div class=\\"v-field__outline\\">
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <div class=\\"v-input__details\\">
      <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\">
        <!---->
      </transition-group-stub>
      <!---->
    </div>
  </div><button type=\\"submit\\" class=\\"v-btn v-btn--block v-btn--elevated v-theme--light bg-primary v-btn--density-default rounded-xl v-btn--size-large v-btn--variant-elevated mt-5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> Загрузить </span>
    <!---->
    <!---->
  </button>
</form>"
`;
